### PR TITLE
MHV-62950: Success/error message still shows when navigating back to Refill page

### PR DESCRIPTION
**Pull request by vmenshutin-bylight**
## Summary

Refill notification data is now cleared from Redux store on Refill page load.

## Related issue(s)

https://jira.devops.va.gov/browse/MHV-62950

image

## Testing done

- Manual testing

## What areas of the site does it impact?

/my-health/medications

## Acceptance criteria

AC1: When navigating back to the refill page after user submits a refill then goes to the list page and back to refill page, the alert should no longer be there. 

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [X] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [X] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/mhv-medications/actions/prescriptions.js [hunk 1]
```diff
@@ -108,3 +108,7 @@ export const fillPrescriptions = prescriptions => async dispatch => {
     return error;
   }
 };
+
+export const clearFillNotification = async dispatch => {
```
Comment (robertbylight) [inline src/applications/mhv-medications/actions/prescriptions.js:112]: would you be able to add a test case for this function?

File: src/applications/mhv-medications/reducers/prescriptions.js [hunk 2]
```diff
@@ -106,6 +106,12 @@ export const prescriptionsReducer = (state = initialState, action) => {
         apiError: false,
       };
     }
+    case Actions.Prescriptions.CLEAR_FILL_NOTIFICATION: {
```
Comment (robertbylight) [inline src/applications/mhv-medications/reducers/prescriptions.js:109]: would you be able to add a test case for this as well?

File: src/applications/mhv-medications/containers/RefillPrescriptions.jsx [hunk 3]
```diff
@@ -117,6 +118,7 @@ const RefillPrescriptions = ({ isLoadingList = true }) => {
   };
 
   useEffect(() => {
+    dispatch(clearFillNotification);
```
Comment (agravell047) [inline src/applications/mhv-medications/containers/RefillPrescriptions.jsx:121]: probably not much of a performance increase but you could add a check to clear only if there's an existing notification

**Review by CBonade — DISMISSED** [review]

File: src/applications/mhv-medications/actions/prescriptions.js (hunk 1 above)
Comment (vmenshutin-bylight) [reply src/applications/mhv-medications/actions/prescriptions.js]: TY. done!

File: src/applications/mhv-medications/reducers/prescriptions.js (hunk 2 above)
Comment (vmenshutin-bylight) [reply src/applications/mhv-medications/reducers/prescriptions.js:109]: TY. done!

File: src/applications/mhv-medications/containers/RefillPrescriptions.jsx (hunk 3 above)
Comment (vmenshutin-bylight) [reply src/applications/mhv-medications/containers/RefillPrescriptions.jsx]: TY. Done!

**Review by robertroddev — DISMISSED** [review]

**Review by agravell047 — DISMISSED** [review]

**Review by agravell047 — APPROVED** [review]

**Review by CBonade — APPROVED** [review]

**Pull request merged**